src/binary/non_blocking/raw_binary_reader.rs

```diff
@@ -475,18 +476,13 @@ impl<A: AsRef<[u8]>> RawBinaryBufferReader<A> {
 
     /// If the reader is currently positioned on a symbol value, parses that value into a `SymbolId`.
     pub fn read_symbol_id(&mut self) -> IonResult<SymbolId> {
-        let (encoded_value, mut buffer) = self.value_and_buffer(IonType::Symbol)?;
-        match buffer.read_uint(encoded_value.value_length())?.value() {
-            UInteger::U64(symbol_id) => {
-                // This will always succeed on 64-bit platforms where u64 and usize are the same.
-                if let Ok(sid) = usize::try_from(*symbol_id) {
-                    Ok(sid)
-                } else {
-                    decoding_error("found a u64 symbol ID that was too large to fit in a usize")
-                }
-            }
-            UInteger::BigUInt(symbol_id) => Self::try_symbol_id_from_big_uint(symbol_id),
```

Comment:
🗺️ This code path (`try_symbol_id_from_big_uint`) would only succeed if the `BigUInt` was small enough that it could have been encoded as a `usize` to begin with. This code path existed solely to satisfy one of the `ion-tests`:
```
ion-tests/iontestdata/good/typecodes/T7-large.10n
```
Text view of its contents:
```
---------------------------------------------------------------------------
 Offset   |  Length   |        Binary Ion        |         Text Ion
---------------------------------------------------------------------------
          |         4 | e0 01 00 ea              |  // Ion 1.0 Version Marker
        4 |         6 | 75 00 00 00 00 00        |   '$0' // $0
       10 |         7 | 76 00 00 00 00 00 00     |   '$0' // $0
       17 |         8 | 77 00 00 00 00 00 00 00  |   '$0' // $0
       25 |         9 | 78 00 00 00 00 00 00 00  |   '$0' // $0
          |           | 00                       |
       34 |        10 | 79 00 00 00 00 00 00 00  |   '$0' // $0
          |           | 00 00                    |
       44 |        11 | 7a 00 00 00 00 00 00 00  |   '$0' // $0
          |           | 00 00 00                 |
       55 |        12 | 7b 00 00 00 00 00 00 00  |   '$0' // $0
          |           | 00 00 00 00              |
       67 |        13 | 7c 00 00 00 00 00 00 00  |   '$0' // $0
          |           | 00 00 00 00 00           |
       80 |        14 | 7d 00 00 00 00 00 00 00  |   '$0' // $0
          |           | 00 00 00 00 00 00        |
       94 |        16 | 7e 8e 00 00 00 00 00 00  |   '$0' // $0
          |           | 00 00 00 00 00 00 00 00  |
```

I have removed that code path altogether and added the test to the skip list. I don't believe this represents a substantive loss of functionality.